subsys/esb/esb.c

```diff
@@ -855,15 +869,34 @@ static void start_tx_transaction(void)
 			retransmits_remaining = esb_cfg.retransmit_count;
 			on_radio_disabled = on_radio_disabled_tx;
 			esb_state = ESB_STATE_PTX_TX_ACK;
+			nrf_radio_int_enable(NRF_RADIO, NRF_RADIO_INT_DISABLED_MASK);
+		} else if (esb_cfg.never_disable_tx) {
+			nrf_radio_shorts_set(NRF_RADIO, radio_shorts_common &
+					~RADIO_SHORTS_END_DISABLE_Msk);
+			nrf_timer_shorts_set(esb_timer.p_reg,
+					(NRF_TIMER_SHORT_COMPARE1_STOP_MASK |
+					NRF_TIMER_SHORT_COMPARE1_CLEAR_MASK));
+			/* Configure timer to produce an ISR after retransmit_delay */
+			nrfx_timer_clear(&esb_timer);
+			nrfx_timer_compare(&esb_timer, NRF_TIMER_CC_CHANNEL1,
+				esb_cfg.retransmit_delay - TRANSMIT_DELAY_US, true);
```

Comment:
The timer is triggered by the PPI once the radio peripheral transitions from TX to TXIDLE (END transition).
So it needs to be fed an inter-transmission delay.
The idea was to allow the user to configure a transmission period (i.e. an interval between transmissions) rather than a delay in-between transmissions.
I've chosen the retransmit_delay variable, as it has no use when this mode is activated (no ACK = no retransmission)
The TRANSMIT_DELAY define was used to compensate tx time. 

**After giving it a second thought, using  TRANSMIT_DELAY has no meaning as its value depends on the payload...**

**==> Would the following be OK: ** 
1. Removal of the TRANSMIT_DELAY define
2. Do you prefer keeping the retransmit_delay config variable to setup an in-between transmission delay (and edit the comment), or create a new configuration variable ?

Reply:
TRANSMIT_DELAY_US has been removed
the use of optimize_txen_delay has been documented